multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java

```diff
@@ -49,13 +51,28 @@ protected StepPhase executeStep(ProcessContext context) {
         return StepPhase.DONE;
     }
 
-    private List<String> computeOldAppsToRename(Set<String> appsToProcess) {
-        return appsToProcess.stream()
-                            .filter(appName -> !BlueGreenApplicationNameSuffix.isSuffixContainedIn(appName))
-                            .collect(Collectors.toList());
+    private List<String> computeOldAppsToRename(DeployedMta deployedMta, List<String> selectedModulesList) {        
+        Set<String> selectedModulesSet = new TreeSet<>();
+        selectedModulesSet.addAll(CollectionUtils.emptyIfNull(selectedModulesList));
+
+        return deployedMta.getApplications()
+                          .stream()
+                          .filter(app -> !BlueGreenApplicationNameSuffix.isSuffixContainedIn(app.getName()))
+                          .filter(app -> moduleIsSelectedForDeployment(app.getModuleName(), selectedModulesSet))
+                          .map(DeployedMtaApplication::getName)
+                          .collect(Collectors.toList());
+    }
+
+    private boolean moduleIsSelectedForDeployment(String module, Set<String> selectedModules) {
```

Comment:
Wouldn't `isModuleSelectedForDeployment` be better?
And rename the first parameter to `moduleName`